src/parquet/arrow/writer.h

```diff
@@ -94,6 +94,12 @@ ::arrow::Status PARQUET_EXPORT WriteTable(const ::arrow::Table& table,
     int64_t chunk_size,
     const std::shared_ptr<WriterProperties>& properties = default_writer_properties());
 
+/**
+ * Helper function for converting nanosecond timestamps to Impala (Int96) format
+ */
+inline void nanosecondsToImpalaTimestamp(const int64_t nanoseconds,
```

Comment:
You can put the whole implementation here, but put it in an `parquet::arrow::internal` namespace. Also use `NanosecondsToImpalaTimestamp`